test/test_case_fixture.h

```diff
@@ -37,6 +37,29 @@ struct test_case_fixture : public base_test_fixture
         double>
         integral_test_types;
 
+    template <typename Functor>
+    void check_database_error_state(
+        Functor operationThrowsDatabaseError,
+        const std::string& expectedState)
+    {
+        bool database_error_thrown = false;
+        try
+        {
+            operationThrowsDatabaseError();
```

Comment:
Should read `operation_throws_database_error`, and `expected_state` too.

Reply:
Thanks, fixed!